crates/router/src/core/payments/helpers.rs

```diff
@@ -3167,6 +3167,7 @@ impl AttemptType {
             merchant_connector_id: None,
             unified_code: None,
             unified_message: None,
+            net_amount: old_payment_attempt.net_amount,
```

Comment:
net_amount and surcharge_details would change for new attempt as payment_method in retries change right?